asv_bench/benchmarks/dataarray_missing.py

```diff
@@ -21,54 +16,65 @@ def make_bench_data(shape, frac_nan, chunks):
     return da
 
 
-def time_interpolate_na(shape, chunks, method, limit):
-    if chunks is not None:
-        requires_dask()
-    da = make_bench_data(shape, 0.1, chunks=chunks)
-    actual = da.interpolate_na(dim="time", method="linear", limit=limit)
-
-    if chunks is not None:
-        actual = actual.compute()
-
-
-time_interpolate_na.param_names = ["shape", "chunks", "method", "limit"]
-time_interpolate_na.params = (
-    [(3650, 200, 400), (100, 25, 25)],
-    [None, {"x": 25, "y": 25}],
-    ["linear", "spline", "quadratic", "cubic"],
-    [None, 3],
-)
-
-
-def time_ffill(shape, chunks, limit):
-
-    da = make_bench_data(shape, 0.1, chunks=chunks)
-    actual = da.ffill(dim="time", limit=limit)
-
-    if chunks is not None:
-        actual = actual.compute()
-
-
-time_ffill.param_names = ["shape", "chunks", "limit"]
-time_ffill.params = (
-    [(3650, 200, 400), (100, 25, 25)],
-    [None, {"x": 25, "y": 25}],
-    [None, 3],
-)
-
-
-def time_bfill(shape, chunks, limit):
-
-    da = make_bench_data(shape, 0.1, chunks=chunks)
-    actual = da.bfill(dim="time", limit=limit)
-
-    if chunks is not None:
-        actual = actual.compute()
-
-
-time_bfill.param_names = ["shape", "chunks", "limit"]
-time_bfill.params = (
-    [(3650, 200, 400), (100, 25, 25)],
-    [None, {"x": 25, "y": 25}],
-    [None, 3],
-)
+def requires_bottleneck():
+    try:
+        import bottleneck  # noqa: F401
+    except ImportError:
+        raise NotImplementedError()
+
+
+class DataArrayMissingInterpolateNA:
+    def setup(self, shape, chunks, limit):
+        if chunks is not None:
+            requires_dask()
+        self.da = make_bench_data(shape, 0.1, chunks)
+
+    @parameterized(
+        ["shape", "chunks", "limit"],
+        (
+            [(100, 25, 25)],
```

Comment:
I think we should keep the big size here.

In general, it'd be nice to have a rule for how big or how long a benchmark should take to be meaningful. 100ms - 500ms maybe? 

cc @pydata/xarray

Reply:
I aimed for 50ms or around 3mins per file. asv becomes unbearably slow with larger sizes and it usually meant that all RAM was eaten up and a complete freeze of my pc. 

I'm not completely convinced about the big sizes but if it is that important then it should be limited to the dask variants only.

Reply:
Here's how long `dataarray_missing.py` takes in this workflow:
321a761 - `shape=(100, 25, 25)`, 3 minutes
1eba65c - `shape=(3650, 200, 400)`, 20 minutes and crashes